ui/app/components/namespace-editor.js

```diff
@@ -0,0 +1,157 @@
+
+      if (
+        this.namespace.isNew &&
+        this.store
+          .peekAll('namespace')
+          .filter((namespace) => namespace !== this.namespace)
+          .findBy('name', this.namespace.name)
+      ) {
+        throw new Error(
+          `A namespace with name ${this.namespace.name} already exists.`
+        );
+      }
+
+      this.namespace.set('id', this.namespace.name);
+      await this.namespace.save();
+
+      this.notifications.add({
+        title: 'Namespace Saved',
+        color: 'success',
+      });
+
+      if (shouldRedirectAfterSave) {
+        this.router.transitionTo(
+          'access-control.namespaces.acl-namespace',
+          this.namespace.name
+        );
+      }
+    } catch (err) {
+      let title = this.namespace.isNew
+        ? `Error creating Namespace ${this.namespace.name}`
+        : `Error updating Namespace ${this.namespace.name}`;
+
+      let message = err.errors?.length
+        ? messageFromAdapterError(err)
+        : err.message || 'Unknown Error';
+
+      this.notifications.add({
+        title,
+        message,
+        color: 'critical',
+        sticky: true,
+      });
+    }
+  }
+
+  definitionStringFromNamespace(namespace) {
+    let definitionHash = {};
+    definitionHash['Description'] = namespace.description;
+    definitionHash['Capabilities'] = namespace.capabilities;
+    definitionHash['Meta'] = namespace.meta;
+
+    if (this.can.can('configure-in-namespace node-pool')) {
+      definitionHash['NodePoolConfiguration'] = namespace.nodePoolConfiguration;
+    }
+
+    if (this.can.can('configure-in-namespace quota')) {
+      definitionHash['Quota'] = namespace.quota;
+    }
+
+    return JSON.stringify(definitionHash, null, 4);
+  }
+
+  deserializeDefinitionJson(definitionHash) {
+    this.namespace.set('description', definitionHash['Description']);
+    this.namespace.set('meta', definitionHash['Meta']);
+
+    let capabilities = this.store.createFragment(
+      'ns-capabilities',
+      definitionHash['Capabilities']
+    );
+    this.namespace.set('capabilities', capabilities);
+
+    if (this.can.can('configure-in-namespace node-pool')) {
+      let npConfig = definitionHash['NodePoolConfiguration'];
+      this.store.create;
+
+      // If we don't manually set this to null, removing
+      // the keys wont update the data framgment, which we want
+      if (!('Allowed' in npConfig)) {
+        npConfig['Allowed'] = null;
+      }
+
+      if (!('Disallowed' in npConfig)) {
+        npConfig['Disallowed'] = null;
+      }
+
+      // Create node pool config fragment
+      let nodePoolConfiguration = this.store.createFragment(
+        'ns-node-pool-configuration',
+        npConfig
+      );
+
+      this.namespace.set('nodePoolConfiguration', nodePoolConfiguration);
+    }
+
+    if (this.can.can('configure-in-namespace quota')) {
+      this.namespace.set('quota', definitionHash['Quota']);
+    }
+  }
```

Comment:
Since we have these as separate fragments etc., do we want to include Quotas, "Allowed rules", "Disallowed rules", etc. as columns on the namespace index?